cli/util/file.go

```diff
@@ -0,0 +1,41 @@
+package util
+
+import (
+	"fmt"
+	"io/fs"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	tfInternalDirPrefix = ".terraform"
+)
+
+// walkTerraformDirs traverses a provided path to return a list of directories
+// that hold terraform configs while skiping internal folders that have a
+// .terraform.* prefix
+func WalkTerraformDirs(topLevelPath string) ([]string, error) {
```

Comment:
I wanted to reuse this in a test discovery function https://github.com/terraform-google-modules/terraform-docs-samples/pull/226. Can we allow a custom skip function as in that PR?

Reply:
Added.